### PR TITLE
Feature/code confirmation screen

### LINKED ISSUE #6: Add registration code confirmation screen design
# REQUIREMENTS
Once the user has received a confirmation code by email, this must be entered on the app.

Input fields for capturing the code and buttons for sending again the code or continuing with the registration process must be shown.

- No functionality is required

# DESIGN
2 3 Registro (Paso 3)

# Acceptance Criteria
- [ ] Implementation must match the provided design

### PR DESCRIPTION
**Pull request by OwenRamire**
# Create the code confirmation screen layout
Close #6 
## Description
- Create the layout of the screen 
- Read the value on the TextInput components
- Clean the form if the user touches the button "Enviar código de nuevo"
- Create the code "1234" to test the screen if the user types the correct code, the incorrect code and if the user doesn't type the code

  ## Screenshots 
- Screen layout: 
Screenshot_1601077944

- Type the correct code: 
CodeConfirmationSuccess

- Type the incorrect code:
CodeConfirmationWrong

- Touch the button "Siguiente" without type the code confirmation:
CodeConfirmationEmpty

### CODE REVIEW

**Review by xErikPsx — DISMISSED** [review]
Everything looks okay Owen, nice!

File: src/screens/ConfirmEmail.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+      fieldFour: '',
+    });
+  };
+
+  const checkCodeConfirmation = () => {
+    if (
+      formData.fieldOne === '' &&
+      formData.fieldTwo === '' &&
+      formData.fieldThree === '' &&
+      formData.fieldFour === ''
```
Comment (luisarmandodv) [inline src/screens/ConfirmEmail.js]: You could use `some()` js function. 
https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Objetos_globales/Array/some

```suggestion
      formData.some(value => value !== '');
```

File: src/screens/ConfirmEmail.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    ) {
+      Alert.alert('type all the code');
+    } else {
+      let userCodeConfirmation = [
```
Comment (luisarmandodv) [inline src/screens/ConfirmEmail.js]: `userCodeConfirmation` could be a constant if you are not modifying its value.

File: src/screens/ConfirmEmail.js [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        formData.fieldOne,
+        formData.fieldTwo,
+        formData.fieldThree,
+        formData.fieldFour,
+      ];
```
Comment (luisarmandodv) [inline src/screens/ConfirmEmail.js]: You could use js destructuration
```suggestion
      let userCodeConfirmation = [
        ...formData
      ];
```

File: src/screens/ConfirmEmail.js [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+      for (let i = 0; i < 4; i++) {
+        if (codeConfirmation[i] === userCodeConfirmation[i]) {
+          auxiliarCounter++;
+        }
+      }
```
Comment (luisarmandodv) [inline src/screens/ConfirmEmail.js]: This is good, but always try to use `forEach` or `map()` in order to make interactions cleaner

File: src/screens/ConfirmEmail.js [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+          <CodeConfirmTxtInput
+            id="fieldThree"
+            value={formData.fieldThree}
+            onChangeText={handleOnChange}
+          />
+          <CodeConfirmTxtInput
+            id="fieldFour"
+            value={formData.fieldFour}
+            onChangeText={handleOnChange}
+          />
```
Comment (luisarmandodv) [inline src/screens/ConfirmEmail.js]: This code is being repeated, make a function where you iterate and return that JSX.

File: src/screens/ConfirmEmail.js [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+          style={styles.nextBtn}
+          onPress={() => checkCodeConfirmation()}>
+          <Text style={styles.nextMessageTxt}>{NEXT_MESSAGE}</Text>
+        </TouchableOpacity>
+      </ScrollView>
+    </SafeAreaView>
+  );
+};
+
+export default ConfirmEmail;
```
Comment (luisarmandodv) [inline src/screens/ConfirmEmail.js:115]: is this component `ConfirmEmail` completely reusable? its children's can be splited in other reusable components.

**Review by luisarmandodv — CHANGES_REQUESTED** [review]

**Comment by luisarmandodv** [comment]
Your commits are pretty well written! Congrats!!

**Review by luisarmandodv — APPROVED** [review]

**Review by xErikPsx — APPROVED** [review]
This looks pretty nice! Good job!

**Pull request merged**